h2quic/client.go

```diff
@@ -178,7 +178,7 @@ func (c *client) RoundTrip(req *http.Request) (*http.Response, error) {
 		requestedGzip = true
 	}
 	// TODO: add support for trailers
-	endStream := !hasBody
+	endStream := !hasBody && req.Method != http.MethodConnect
```

Comment:
Emm, my understanding is below,

The statement `req.Method != http.MethodConnect` here does not mean there's a body with request.

It just enforce 'enStream = true' during a CONNECT request. 
So h2quic.Server here https://github.com/lucas-clemente/quic-go/blob/master/h2quic/server.go#L190-L195 will say, "OK, I will not close datastream for reading, maybe the client will do a tls handshake through me".

Please correct me any misunderstanding.